javascript/javascript1/week4/homework/voice-assistant.js

```diff
@@ -0,0 +1,172 @@
+// Analyses the input voice command and dispatches it to the appropriate processor function
+function getReply(command){
+    if (command.startsWith("Hello my name is ")) {
+        replyToHello(command);
+    }
+    else if (command == "What is my name?") {
+        replyWithLastSavedName();
+    }
+    else if (command.startsWith("Add ") && command.endsWith(" to my todo")) {
+        addToTodoList(command);
+    }  
+    else if (command.startsWith("Remove ") && command.endsWith(" from my todo")) {
+        removeFromTodoList(command);
+    }
+    else if (command == "What's on my todo?") {
+        getTodoList();
+    }
+    else if (command == "What day is it today?") {
+        getTodayFormattedDate();
+    }
+    else if (command.startsWith("what is ")) {
+        doSimpleMath(command);
+    }
+    else if (command.startsWith("Set a timer for ") && command.endsWith(" minutes")) {
+        setTimer(command);
+    }
+}
+
+let namesList = [];
+function replyToHello(command){
+    let nameOfUser = command.slice(17).trim();
+    if (nameOfUser == ""){
+        console.log("No name was defined! Please add your name.");
+        return;
+    }
+    for(i=0; i < namesList.length; i++){
+        if (nameOfUser == namesList[i]){
+            console.log("The name is already used!");
+            return;
+        }
+    }
+    namesList.push(nameOfUser);
+    console.log("Nice to meet you " + nameOfUser);
+}
+
+function replyWithLastSavedName(){
+    if (namesList == null) {
```

Comment:
I think this is not sufficient if the list is empty, because that is not null but you can't remove from an empty list

Reply:
Thanks for spotting it, Florin. I will change accordingly.